packages/read-model/tests/Unit/Image/ImageViewRepositoryTest.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\ReadModelBundle\Unit\Image;
+
+use PHPUnit\Framework\TestCase;
+use Shopsys\FrameworkBundle\Component\Image\Image;
+use Shopsys\FrameworkBundle\Component\Image\ImageFacade;
+use Shopsys\ReadModelBundle\Image\ImageView;
+use Shopsys\ReadModelBundle\Image\ImageViewFactory;
+use Shopsys\ReadModelBundle\Image\ImageViewRepository;
+
+class ImageViewRepositoryTest extends TestCase
+{
+    private const IMAGE_EXTENSION = 'jpg';
+
+    /** @var \Shopsys\FrameworkBundle\Component\Image\ImageFacade */
+    private $imageFacadeMock;
+
+    protected function setUp()
+    {
+        parent::setUp();
+
+        $this->imageFacadeMock = $this->createMock(ImageFacade::class);
+        $this->imageFacadeMock->method('getImagesOrNullsByEntitiesIndexedByEntityId')
+            ->willReturnCallback(function ($entityIds, string $entityClass) {
+                $images = [];
+
+                foreach ($entityIds as $entityId) {
+                    // id 800 represents not existing product
+                    if ($entityId === 800) {
+                        continue;
+                    }
+
+                    $images[$entityId] = $this->createImageMock($entityId, $entityClass);
+                }
+
+                return $images;
+            });
+    }
+
+    public function testGetForEntityIds(): void
+    {
+        $imageFactory = new ImageViewFactory();
+        $imageViewRepository = new ImageViewRepository($this->imageFacadeMock, $imageFactory);
+
+        $imageViews = $imageViewRepository->getForEntityIds('product', [1, 3, 5]);
+
+        $this->assertEquals(new ImageView(1, self::IMAGE_EXTENSION, 'product', null), $imageViews[1]);
```

Comment:
you can create the whole expected array...
```php
$expected = [
1 => new ImageView(1, self::IMAGE_EXTENSION, 'product', null),
3 => new ImageView(3, self::IMAGE_EXTENSION, 'product', null),
5 => new ImageView(5, self::IMAGE_EXTENSION, 'product', null),
];
$this->assertEquals($expected, $imaveViews);
```
... what do You think? What is more readable? (also this option checks number of elements);

Reply:
Nicer, thanks :)